server-app/src/persistence/sample_DAO.py

```diff
@@ -15,17 +16,45 @@ def createSample(sampleJson):
     if((new_sample.sourceSampleId != 0 and existing_sample_id) or existing_sample_name):
         return 0
     else:
-        new_sample.commit()
-        new_sample.dump()
-        print("Created: " + new_sample.name)
-        return Sample.find_one({"id": new_sample.id})
+        created = new_sample.commit()
+        return Sample.find_one({"id": created.inserted_id}).dump()
+
+
+def createIndividualSample(sampleJson):
+    new_sample = Sample(**sampleJson)
+
+    existing_parent_sample = Sample.find_one(
+        {"id": ObjectId(new_sample.parentSampleId)})
+    existing_sample_name = Sample.find_one({"name": new_sample.name})
```

Comment:
i'm not sure i can understand why do we need to query here based on the sample name